databroker/_core.py

```diff
@@ -1119,7 +1119,14 @@ def add_filter(self, **kwargs):
         :meth:`Broker.clear_filters`
 
         """
-        self.filters.append(dict(**kwargs))
+        for k,v in kwargs.items():
```

Comment:
I'm not sure. This feels like second-guessing how the language works. You are allowed to do `d = {'a': 1}; d['a'] = 2` without doing `del d['a']` in the middle. I'm not convinced filters should work differently.